src/test/compile-fail/associated-types-outlives.rs

```diff
@@ -27,6 +27,7 @@ fn denormalise<'a, T>(t: &'a T) -> <T as Foo<'a>>::Bar {
 pub fn free_and_use<T: for<'a> Foo<'a>,
                     F: for<'a> FnOnce(<T as Foo<'a>>::Bar)>(x: T, f: F) {
     let y;
+    #[allow(unused_labels)]
```

Comment:
@KyleStach1678 
Test modifications should be unnecessary because compile-fail and UI tests allow `unused` warnings by default.
So, instead `unused_labels` needs to be added to the `unused` lint group here https://github.com/rust-lang/rust/blob/ed9a29a976f8b73f3fc91a10d0b5addd86a394c7/src/librustc_lint/lib.rs#L162-L180.